Tetris Okhrimenko v1.0

napstergit1 opened the pull request:
Из проблем на данный момент:
1) Нужно показывать следующую фигуру
2) Исправить код стайл
3) Проверить код на наличие лишнего  "мусора"
4) Корректно обрабатывать проигрыш(сейчас немного не в тот момент срабатывает сигнал)

napstergit1 committed:
Tetris Okhrimenko v1.0

Из проблем на данный момент:
1) Нужно показывать следующую фигуру
2) Исправить код стайл
3) Проверить код на наличие лишнего  "мусора"
4) Корректно обрабатывать проигрыш(сейчас немного не в тот момент срабатывает сигнал)

hBuzzy commented on gamefield.h:
```diff
@@ -8,29 +8,86 @@ class GameField : public QWidget {
+
+    bool CheckCollisionMoveLeft();
+    bool CheckCollisionMoveRight();
+    bool CheckCollisionRotate();
```
Либо все методы с большой (как принято в код стиле у нас), либо все с маленькой, придерживаясь стиля QT. 

Check - всегда плохое слово, которое почти никогда достаточно не погружает в контекст того, что мы проверяем. Кроме того, методы / переменные / поля типа bool должны задавать вопрос, на который мы можем ответить строго "Да" или "Нет". У вас вопросов нет вообще, только утвердительные формы.

В создании вопроса нам помогают Is, Has, Have и т.д.  CheckCollision -> HasCollisions()  и т.д.

hBuzzy commented on gamefield.h:
```diff
@@ -8,29 +8,86 @@ class GameField : public QWidget {
+    bool CheckCollisionRotate();
+
+    QVector<QVector<int>> GetCurrentFigure();
+    QVector<QVector<int>> GetRotateCurrentFigure();
```
Rotate - глагол.

Если вы хотите получить поворот, то это rotation. Плюс нужно соблюдать порядок слов. GetCurrentFigureRotation. Технически, конечно, стоит все же выделить ваши наборы фигур в класс фигура хотя бы, чтобы не было такой каши в названиях. 

Тогда бы у вас было что-то около: `Figure::GetRotation`.

hBuzzy commented on gamefield.h:
```diff
@@ -8,29 +8,86 @@ class GameField : public QWidget {
+
+    QVector<QVector<int>> currentFigure_;
+
+    QVector<QVector<int>> GameGrid_; // Поле с информацией о заполнении поля фигурами
```
С маленькой буквы

hBuzzy commented on gamefield.h:
```diff
@@ -8,29 +8,86 @@ class GameField : public QWidget {
+
+    QVector<QVector<int>> GameGrid_; // Поле с информацией о заполнении поля фигурами
+
+    uint currentFigureRow_;
+    uint currentFigureColumn_;
+
```
Опять же, выделив все хотя бы в класс фигура, такой каши не будет.

hBuzzy commented on gamefield.h:
```diff
@@ -8,29 +8,86 @@ class GameField : public QWidget {
+    uint currentFigureRow_;
+    uint currentFigureColumn_;
+
+    QColor colorFigure;
```
`QColor colorFigure; `-> `QColor figureColor_;`

hBuzzy commented on gameoverdialog.h:
```diff
@@ -0,0 +1,22 @@
+
+private:
+
+    QPushButton *newTryButton;
+    QLabel *label;
```
Постфикс "_" у приватных полей.

hBuzzy commented on tetris.cpp:
```diff
@@ -1,9 +1,153 @@
+    lcdNumber->setFixedSize(200,50);
+    lcdNumber->setDigitCount(6);
+    lcdNumber->display("000000");
+
+
```
Не больше двух пустых строк

hBuzzy commented on tetris.cpp:
```diff
@@ -1,9 +1,153 @@
+
+
+    gamefield = new GameField(this);
+    gamefield->setFixedSize(601,721);
```
Пробел после запятой

hBuzzy commented on tetris.cpp:
```diff
@@ -1,9 +1,153 @@
+
+    QGridLayout *gridLayout = new QGridLayout;
+
+    gridLayout->addWidget(labelPoints,1, 0);
```
Пробел после запятой.

hBuzzy commented on tetris.cpp:
```diff
@@ -1,9 +1,153 @@
+    this->setMinimumSize(1200, 800);
+    this->setMaximumSize(1200, 800);
+
+    lcdNumber = new QLCDNumber;
```
Слишком общее название, уточните: что это? Если счётчик очков, то так и напишите.

hBuzzy commented on tetris.cpp:
```diff
@@ -1,9 +1,153 @@
+
+    gamefield->clearGameGrid();
+    gamefield->spawnNextFigure();
+    gamefield->SetFigurePosition(0, gamefield->GetColumnsNumber() / 2);
```
2 - магическое число

hBuzzy commented on tetris.h:
```diff
@@ -16,7 +21,24 @@ class Tetris : public QMainWindow {
  private:
   Ui::Tetris *ui;
+
+    QTimer *timer;
+    GameField *gamefield;
+    ModalDialog *modaldialog;
+    QLCDNumber *lcdNumber;
+    void openModalDialog();
+    void openEndGameDialog();
+
+    void keyPressEvent(QKeyEvent *event);
+
```
Сначала приватные методы, только потом приватные поля.

void keyPressEvent(QKeyEvent *event) - точно должен быть приватным, а не Protected?

У полей постфикс "_"

hBuzzy commented on tetris.h:
```diff
@@ -16,7 +21,24 @@ class Tetris : public QMainWindow {
+
+    QTimer *timer;
+    GameField *gamefield;
+    ModalDialog *modaldialog;
```
Слишком общее название, уточните.

hBuzzy commented on tetris.cpp:
```diff
@@ -1,9 +1,153 @@
+    dialog.exec();
+    timer->start(400);
+}
+
+
+
+void Tetris::openEndGameDialog() {
```
Не больше одной пустой строки

hBuzzy commented on gamefield.cpp:
```diff
@@ -1,32 +1,373 @@
+        }
+    }
+
+    for (int i = 0; i < rowsNumber_;i++) {
+        for (int j = 0; j < columnsNumber_;j++) {
```
Пробелы после ; в условиях

hBuzzy commented on gamefield.cpp:
```diff
@@ -1,32 +1,373 @@
+    if(!isGameOver) {
+        // Проверка, не касается ли текущая фигура нижней границы поля
+        if (currentFigureRow_ + currentFigure_.size() < rowsNumber_) {
+            bool isHaveCollision = CheckCollision();
```
`isHaveCollision` -> `hasCollision`(-s)

hBuzzy commented on gamefield.cpp:
```diff
@@ -1,32 +1,373 @@
+                CheckLine();
+                //CheckColumn();
+                GameGrid_[x][y] = 1;
+            }
+        }
+    }
+}
+
+bool GameField::CheckCollision() {
+    for (int i = 0; i < currentFigure_.size(); i++) {
+        for (int j = 0; j < currentFigure_[0].size(); j++) {
+            if (currentFigure_[i][j] == 1) {
+                int x = currentFigureRow_ + i + 1;
+                int y = currentFigureColumn_ + j;
+
+                if (x < rowsNumber_ && GameGrid_[x][y] == 1) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+
+bool GameField::CheckCollisionMoveLeft() {
+    for (int i = 0; i < currentFigure_.size(); i++) {
+        for (int j = 0; j < currentFigure_[0].size(); j++) {
+            if (currentFigure_[i][j] == 1) {
+                int x = currentFigureRow_ + i;
+                int y = currentFigureColumn_ + j - 1;
+
+                if (y < 0 || (x < rowsNumber_ && GameGrid_[x][y] == 1)) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
```
Все 3 методы передвижения (проверки передвижения?) - дубляж кода. Выделите общую логику в отдельный метод.

hBuzzy reviewed (commented):
Исправить замечания. Убрать ненужные классы, желательно все же выделить фигуры хотя бы в один отдельный класс. Убрать комментарии.

napstergit1 committed:
Okhrimenko Tetris v2.0

Исправлены замечания.
Корректное отображение следующей фигуры, очков и всего что необходимо по заданию.

napstergit1 reviewed (commented):
Версия 2
